src/http/v4.rs

```diff
@@ -0,0 +1,94 @@
+use std::time::Duration;
+
+use cloudflare::framework::auth::Credentials;
+use cloudflare::framework::response::ApiFailure;
+use cloudflare::framework::{Environment, HttpApiClient, HttpApiClientConfig};
+use http::StatusCode;
+
+use crate::http::{feature::headers, Feature};
+use crate::settings::global_user::GlobalUser;
+use crate::terminal::{emoji, message};
+
+pub struct CfApiClientConfig {
+    feature: Option<Feature>,
+    http_timeout: Duration,
+}
+
+impl CfApiClientConfig {
+    pub fn default() -> HttpApiClientConfig {
```

Comment:
this isn't the usual signature of default; Default is a Trait where the interface is a function default that returns the struct it is impl'd on; so here it could be kind of  confusing. i'm inclined to simplify into bare functions that return an HttpApiClientConfig which includes our default headers. Since HttpApiClientConfig already has a field for timeout, it feels weird to do this if what we're trying to standardize is the headers. i'd be happy writing

```rust
let http_config = HttpApiConfig {
  default_headers: wrangler_headers(Some(feature))
  ..Default::default()
}}
```
or mutating a default struct:

```rust
let mut http_config = wrangler_default_api_config();
add_feature(&http_config, FEATURE);
```
until we need something more clever. this pattern feels like overkill here atm.

Reply:
ooh - TIL that Rust has a spread operator like JavaScript. Definitely gonna use that instead